module/zfs/dbuf.c

```diff
@@ -3488,7 +3492,7 @@ dbuf_prefetch_impl(dnode_t *dn, int64_t level, uint64_t blkid,
 		zbookmark_phys_t zb;
 
 		/* flag if L2ARC eligible, l2arc_noprefetch then decides */
-		if (dnode_level_is_l2cacheable(&bp, dn, level))
+		if (dpa->dpa_aflags & ARC_FLAG_L2CACHE)
 			iter_aflags |= ARC_FLAG_L2CACHE;
```

Comment:
And this is a cache-ability of intermediate indirect blocks.  It may be different from one of the actual block requested for prefetch.  So I am not sure this is correct.